lib/reek/source/source_code.rb

```diff
@@ -38,9 +41,9 @@ def initialize(code, description, parser = Parser::Ruby22)
       def self.from(source)
         case source
         when File     then new(source.read, source.path)
-        when IO       then new(source.readlines.join, 'STDIN')
+        when IO       then new(source.readlines.join, IO_IDENTIFIER)
         when Pathname then new(source.read, source.to_s)
-        when String   then new(source, 'string')
+        when String   then new(source, STRING_IDENTIFIER)
         end
```

Comment:
@troessner: fair enough. :)

@mvz: I was thinking just that yesterday when I was pondering on #604, but the more I tried to describe the `Source` object (that has an origin and can be read from, and is subclassed for the different subtypes), the more it matched the `SourceCode` object we already have, so maybe we can make that a bit more elastic?